src/Database/Relational.php

```diff
@@ -1534,33 +1535,112 @@ public function destroy()
 	/**
 	 * Checks out the current model to the provided user
 	 *
-	 * @param   string  $userId  Optional userId for whom the row should be checked out
-	 * @return  $this
+	 * @param   integer  $userId  Optional userId for whom the row should be checked out
+	 * @return  boolean
 	 * @since   2.0.0
 	 **/
 	public function checkout($userId = null)
 	{
-		$userId = $userId ?: \User::get('id');
-		$this->set('checked_out', $userId)
-		     ->set('checked_out_time', \Date::toSql())
-		     ->save();
+		if (!$this->isNew())
+		{
+			$columns = $this->getStructure()->getTableColumns($this->getTableName());
+
+			$data = [];
+
+			if (isset($columns['checked_out_time']))
+			{
+				$now = new Date('now');
+				$data['checked_out_time'] = $now->toSql();
+			}
+
+			if (isset($columns['checked_out']))
+			{
+				$userId = $userId ?: \User::get('id');
+				$data['checked_out'] = $userId;
+			}
+
+			if (empty($data))
+			{
+				// There is no 'checked_out_time' or 'checked_out' column
+				return true;
+			}
+
+			$this->set($data);
+
+			// We build a simple update query as calling save()
+			// can have unintended consequences when all we want
+			// is to update two columns
+			$query = $this->getQuery()
+				->update($this->getTableName())
+				->set($data)
+				->whereEquals($this->getPrimaryKey(), $this->get($this->getPrimaryKey()));
+
+			// @FIXME: Maybe unnecessary? Database may throw an exception on error
+			//         so this might be pointless.
+			if (!$query->execute())
+			{
+				$this->addError(__CLASS__ . '::' . __METHOD__ . '() failed');
+				return false;
```

Comment:
Should we reset the check in/out values on $this here? ( this->checkout would return false, but the object would still contain the checkout information)

Reply:
Good point. I'll make the change (though, I'm still uncertain if it would ever hit that point).